packages/driver/src/cypress/request-logs.ts

```diff
@@ -254,44 +240,40 @@ export default class ProxyLogging {
           proxyRequest.log.end()
         }
       }
-      this.unloggedPreRequests = []
       this.proxyRequests = []
     })
   }
 
-  /**
-   * Update an existing proxy log with an interception, or create a new log if one was not created (like if shouldLog returned false)
-   */
-  logInterception (interception: Interception, route: Route): ProxyRequest {
-    const unloggedPreRequest = take(this.unloggedPreRequests, ({ requestId }) => requestId === interception.browserRequestId)
-
-    if (unloggedPreRequest) {
-      debug('interception matched an unlogged prerequest, logging %o', { unloggedPreRequest, interception })
-      this.createProxyRequestLog(unloggedPreRequest)
+  set filter (filterFn: typeof this._filter | undefined) {
+    if (filterFn !== undefined && typeof filterFn !== 'function') {
+      $errUtils.throwErrByPath('network_logs.filter_must_be_function', { args: { filterFn } })
     }
 
+    this._filter = filterFn
+  }
+
+  get filter () {
+    return this._filter
+  }
+
+  /**
+   * Returns a setter which can be used to update the request's log as `cy.intercept()` rules run.
+   */
+  getFlagSetter (interception: Interception, route: Route): RequestLog['setFlag'] | undefined {
```

Comment:
If you simplify the return type of this function so it always returns a function - a noop if nothing else - then we could also avoid all the `.?()`s you had to add across the codebase like `request.setLogFlag?.('resModified')`.

Reply:
Not a bad idea. Done in a79b9435f01bcfa69f20348b001a449164ae91ae